metsrw/plugins/yapremisrw/event.py

```diff
@@ -0,0 +1,291 @@
+"""PREMIS-Reader-Writer 2 Event
+
+Supplies a PREMIS ``Event`` class for reading and writing PREMIS events.
+Supports the API required for a metsrw plugin, i.e., provides the following
+methods:
+
+1. A ``fromtree`` class method that parses an lxml element and returns a Python
+   instance.
+
+2. A ``serialize`` method that returns an lxml element.
+
+The ``metsrw.FSEntry`` class can accept ``yapremisrw.Event`` as a dependency. To
+inject it::
+
+    >>> from metsrw import feature_broker
+    >>> from metsrw.plugins import yapremisrw
+    >>> feature_broker.provide('premis_event_class', yapremisrw.Event)
+
+"""
+
+from collections import OrderedDict
+from copy import deepcopy
+from lxml import etree
+
+from .exceptions import ConstructError, ParseError
+from . import utils
+
+
+class Event(object):
+    """An object representing a PREMIS event element.
+
+    :param str identifier_type: Identifier type.
+    :param str identifier_value: Identifier value.
+    :param str event_type: Event type.
+    :param str event_datetime: Event date/time.
+    :param str detail: Event details.
+    :param list outcomes: List of dicts containing outcome data.
+    :param list linking_agent_identifiers: List of dicts containing linking
+        agent identifier data.
+    :param list linking_object_identifiers: List of dicts containing linking
+        object identifier data.
+    :raises exceptions.ParseError: If the root tag is not event.
+    """
+    def __init__(self, identifier_type, identifier_value, event_type,
+                 event_datetime, detail=None, outcomes=None,
+                 linking_agent_identifiers=None,
+                 linking_object_identifiers=None):
+
+        if identifier_type is None:
+            raise ConstructError('identifier_type argument is required.')
+
+        if identifier_value is None:
+            raise ConstructError('identifier_value argument is required.')
+
+        if event_type is None:
+            raise ConstructError('event_type argument is required.')
+
+        if event_datetime is None:
```

Comment:
Comment by @Hwesta: This feels unpythonic - these parameters are required by the function, and if they're None, it will raise an error later.